spdx/writers/xml.py

```diff
@@ -9,22 +9,185 @@
+            sha1 = None
+            for checksum in phile.checksums:
+                if checksum.identifier == 'SHA1':
+                    sha1 = checksum.value
+                    break
+            add_not_empty(file_element, 'sha1', sha1)
+            add_not_empty(file_element, 'name', phile.name)
+            add_not_empty(file_element, 'copyrightText', phile.copyright)
+            if len(phile.artifact_of_project_name) > 0 or \
+                    len(phile.artifact_of_project_home) > 0 or \
+                    len(phile.artifact_of_project_uri) > 0:
+                artifact_of_element = ET.SubElement(file_element, 'artifactOf')
+                for apn in phile.artifact_of_project_name:
+                    add_not_empty(artifact_of_element, 'name', apn)
+                for aph in phile.artifact_of_project_home:
+                    add_not_empty(artifact_of_element, 'homePage', aph)
+                for apu in phile.artifact_of_project_uri:
+                    add_not_empty(artifact_of_element, 'projectUri', apu)
+            add_license(file_element, 'licenseConcluded', phile.conc_lics)
+            add_not_empty(file_element, 'licenseComments', phile.license_comment)
+            for checksum in phile.checksums:
+                add_checksum(file_element, 'checksum', checksum)
+            for file_type in phile.file_types:
+                file_type_name = file.FILE_TYPE_TO_XML_DICT.get(file_type)
+                if file_type_name is None:
+                    raise RuntimeError('unknown file type {}'.format(file_type))
+                add_not_empty(file_element, 'fileType', file_type_name)
+            add_not_empty(file_element, 'SPDXID', phile.spdx_id)
+
+        for license_from_file in package.licenses_from_files:
+            add_not_empty(packages_element, 'licenseInfoFromFiles', license_from_file.identifier)
+        add_not_empty(packages_element, 'sha1', package.check_sum.value)
+        add_not_empty(packages_element, 'name', package.name)
+        add_not_empty(packages_element, 'packageFileName', package.file_name)
+        add_not_empty(packages_element, 'licenseComments', package.license_comment)
+        add_not_empty(packages_element, 'summary', package.summary)
+        add_not_empty(packages_element, 'sourceInfo', package.source_info)
+        add_not_empty(packages_element, 'copyrightText', package.cr_text)
+        package_verification_code_element = ET.SubElement(packages_element, 'packageVerificationCode')
+        add_not_empty(package_verification_code_element, 'packageVerificationCodeValue', package.verif_code)
+        for verif_exc_file in package.verif_exc_files:
+            add_not_empty(package_verification_code_element, 'packageVerificationCodeExcludedFiles', verif_exc_file)
+        add_license(packages_element, 'licenseConcluded', package.conc_lics)
+        add_not_empty(packages_element, 'supplier', package.supplier)
+        add_checksum(packages_element, 'checksum', package.check_sum)
```

Comment:
As for license concluded, checksum is optional: https://spdx.github.io/spdx-spec/v2.3/package-information/#710-package-checksum-field